src/main/java/book/store/onlinebookstore/repository/book/BookRepository.java

```diff
@@ -1,8 +1,20 @@
 package book.store.onlinebookstore.repository.book;
 
 import book.store.onlinebookstore.model.Book;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.EntityGraph;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
 
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+    @EntityGraph(attributePaths = "categories")
+    Optional<Book> findById(Long id);
+
+    @EntityGraph(attributePaths = "categories")
+    Page<Book> findAll(Pageable pageable);
+
+    List<Book> findAllByCategoriesId(Long id, Pageable pageable);

```

Comment:
will this work at all? have you tested it?

Reply:
yes